server/honeybadger/plugins.py

```diff
@@ -26,18 +24,43 @@ def get_coords_from_google(aps):
         data['lng'] = jsondata['location']['lng']
     return data
 
-def get_coords_from_uniapple(ip):
-    logger.info('Geolocating via Uniapple API.')
-    url = 'http://uniapple.net/geoip/?ip={}'.format(ip)
-    content = urllib2.urlopen(url).read()
-    logger.info('Uniapple API response:\n{}'.format(content))
+def get_coords_from_ipstack(ip):
+    logger.info('Geolocating via Ipstack API.')
+    url = 'http://api.ipstack.com/{0}?access_key={1}'.format(ip, app.config['IPSTACK_API_KEY'])
+    content = requests.get(url).content
+    logger.info('Ipstack API response:\n{}'.format(content))
     jsondata = None
     try:
         jsondata = json.loads(content)
     except ValueError as e:
         logger.error('{}.'.format(e))
+
+    # Avoid the KeyError. For some reason, a successful API call to Ipstack doesn't include
+    #   the 'success' key in the json result, but a failed call does, and the value is False
+    if 'success' in jsondata and not jsondata['success']:
+        logger.info('Ipstack API call failed: {}'.format(jsondata['error']['type']))
+        # Return with no data so the caller knows to default to the fallback API
+        return None
+
     data = {'lat':None, 'lng':None}
     if jsondata:
         data['lat'] = jsondata['latitude']
         data['lng'] = jsondata['longitude']
     return data
+
+def get_coords_from_ipinfo(ip):
+    # New fallback, ipinfo doesn't require an API key for a certain number of API calls
+    logger.info('Geolocating via Ipinfo.io API.')
+    url = 'https://ipinfo.io/{}'.format(ip)
+    content = requests.get(url).content
+    logger.info('Ipinfo.io API response:\n{}'.format(content))
+    jsondata = None
+    try:
+        jsondata = json.loads(content)
+    except ValueError as e:
+        logger.error('{}.'.format(e))
+    data = {'lat':None, 'lng':None}
```

Comment:
Re: the above comment, this is what the rest of these functions return.

Reply:
Fair enough, so I suppose I should follow this convention in my change. I'll take care of it when I get a chance.

Reply:
If it makes sense. Like I said above, I need to look at where these are getting called and see what it does exactly. Been a while since I wrote that code.